scipy/optimize/_basinhopping.py

```diff
@@ -310,7 +310,16 @@ def accept_reject(self, energy_new, energy_old):
         If new is higher than old, there is a chance it will be accepted,
         less likely for larger differences.
         """
-        w = math.exp(min(0, -float(energy_new - energy_old) * self.beta))
+        # energy_new, energy_old are sometimes scalars, sometimes arrays of
+        # length 1. Cast them all into scalars.
+        energy_new = np.asarray(energy_new).reshape(1)[0]
+        energy_old = np.asarray(energy_old).reshape(1)[0]
+        # Not casting energy_diff into a Python float reveals an error in the tests:
+        # The expression 0 * inf is undefined. With Python floats this simply gives
```

Comment:
That's not true, it's defined as nan in numpy too - it just emits a warning at the same time.

Reply:
Oh yes, you're right, it's just a warning. Tests with will fail as warnings apparently aren't permitted:
```
E       RuntimeWarning: invalid value encountered in double_scalars
```

Reply:
I suspect the fix is to suppress the warning in the test, since the test is probably checking what happens when questionable values are used.